server.py

```diff
@@ -5,11 +5,575 @@
+        "rating": 5,
+        "comments": "Head to this laid-back Long Island City restaurant for beef tongue tacos, chicken smothered in a heady mole sauce, and a monster crab tostada. New York's only Michelin-starred Mexican restaurant is an especially cool choice for lunch during the week or drinks after work. Eater critic Ryan Sutton awarded this restaurant two stars."
+      },
+      {
+        "id": 29,
+        "restaurant_id": 10,
+        "name": "Rob",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 5,
+        "comments": "The hype was real. Please go. Get the ceviche. And the tres leches. You're welcome"
+      },
+      {
+        "id": 30,
+        "restaurant_id": 10,
+        "name": "Jason",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 4,
+        "comments": "For a Michelin star restaurant, it's fairly priced and the food is fairly good. Started with a strawberry margarita which was good in flavor but not much alcohol. Had the chicken enchiladas with salsa verde and it was really good. Great balance in flavor and a good portion. Extra tasty with their hot sauces. My wife had the lamb but it was a bit too salty for our taste. Although, it was cooked very well and fell off the bone. The highlight of the night was the tres leches cake. Probably the best I've ever had to be honest. Not too sweet and very milky. Overall, one of my top 3 favorite Mexican in NY."
+      }]
+
+
+@app.route('/restaurants', methods=['GET'])
+def getAllRestaurants():
+    return jsonify(restaurants)
+
+@app.route('/reviews', methods=['GET'])
+def getAllReviews():
+    return jsonify(reviews)
+
+@app.route('/restaurants/favorites', methods=['GET'])   
+def getFavoritedRestaurants():
+  results = []
+  for restaurant in restaurants:
+    if(restaurant.get("is_favorite") == True):
+      results.append(restaurant)
+  return jsonify(results)    
+
+@app.route('/restaurants/<int:restaurant_id>', methods=['GET']) 
+def getRestaurantByID(restaurant_id):
+  for restaurant in restaurants:
+    if(restaurant.get("id") == restaurant_id):
+      return jsonify(restaurant)  
+
+@app.route('/reviews/restaurant/<int:restaurant_id>', methods=['GET'])
+def getAllReviewsForRestaurant(restaurant_id):
+  results = []
+  for review in reviews:
+    if(review.get("restaurant_id") == restaurant_id):
+      results.append(review)
+  return jsonify(results) 
+
+@app.route('/reviews/remove/<int:review_id>', methods=['GET'])
+def removeRestaurantReview(review_id):
+  review_id = request.json.get("id")
+  for review in reviews:
+    if(review.get("id") == review_id):
+      reviews.remove(review)
+      break
+  return jsonify(reviews)  
+
+@app.route('/reviews/<restaurant_id>', methods=['POST'])
+def addRestaurantReview(restaurant_id):
+  if request.headers['Content-Type'] == 'application/json':
+    review = request.json
+    review["id"] = len(reviews) + 1
```

Comment:
This might not really matter here, but this is a good learning opportunity:
This is dangerous code, since if two different requests come in at the same time, they will likely end up with the same review[id]

Reply:
I was aware that this wasn't ideal. Especially once you start deleting reviews, the length would not be an accurate way of figuring out the last id. (Or if you import data and the data starts with non contigious id numbers.) This was more of a placeholder while I figure something else out.

It seems like I could use UUID which from my understanding doesn't prevent duplicates eventually or I could just start having the reviews and restaurants as objects and create methods to convert them into json when needed. 

I was thinking about trying to finish the other methods before doing this -> switching to object based. Thoughts?

Reply:
Uuid is good here, objects won't get u this functionality